packages/block-editor/src/components/rich-text/index.native.js

```diff
@@ -817,7 +817,13 @@ RichText.defaultProps = {
 
 const RichTextContainer = compose( [
 	withInstanceId,
-	withBlockEditContext( ( { clientId } ) => ( { clientId } ) ),
+	withBlockEditContext( ( { clientId, onFocus, isSelected }, ownProps ) => {
+		return { 
+			clientId: clientId,
+			isSelected: isSelected,
```

Comment:
isSelected seems to be always undefined on withSelect. Passing it on withBlockEditContext will work and fix ListBlock split.